shared/reducers/chat2.js

```diff
@@ -835,9 +841,30 @@ const rootReducer = (
       const {conversationIDKey, messageID, requestInfo} = action.payload
       return state.update('accountsInfoMap', old => old.setIn([conversationIDKey, messageID], requestInfo))
     }
+    case Chat2Gen.attachmentFullscreenSelection: {
+      const {message} = action.payload
+      return state.set('attachmentFullscreenMessage', message)
+    }
     case Chat2Gen.handleSeeingWallets: // fallthrough
     case Chat2Gen.setWalletsOld:
       return state.isWalletsNew ? state.set('isWalletsNew', false) : state
+    case Chat2Gen.attachmentDownloaded:
+      const {message, path} = action.payload
+      let nextState = state
+      // check fullscreen attachment message in case we downloaded it
+      if (
+        state.attachmentFullscreenMessage &&
+        state.attachmentFullscreenMessage.conversationIDKey === message.conversationIDKey &&
+        state.attachmentFullscreenMessage.id === message.id &&
+        message.type === 'attachment'
+      ) {
+        nextState = nextState.set('attachmentFullscreenMessage', message.set('downloadPath', path))
+      }
+      return nextState.withMutations(s => {
+        s.set('metaMap', metaMapReducer(state.metaMap, action))
```

Comment:
can use merge and remove withMutations/set

Reply:
I just copy/pasted this up from how it was being handled before.